src/client/common/experiments/service.ts

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { inject, named } from 'inversify';
+import { Memento } from 'vscode';
+import { getExperimentationService, IExperimentationService, TargetPopulation } from 'vscode-tas-client';
+import { sendTelemetryEvent } from '../../telemetry';
+import { EventName } from '../../telemetry/constants';
+import { IApplicationEnvironment } from '../application/types';
+import { GLOBAL_MEMENTO, IConfigurationService, IExperimentService, IMemento, IPythonSettings } from '../types';
+import { ExperimentationTelemetry } from './telemetry';
+
+export class ExperimentService implements IExperimentService {
+    /**
+     * Experiments the user requested to opt into manually.
+     */
+    public _optInto: string[] = [];
+    /**
+     * Experiments the user requested to opt out from manually.
+     */
+    public _optOutFrom: string[] = [];
+
+    private readonly experimentationService?: IExperimentationService;
+    private settings!: IPythonSettings;
```

Comment:
```suggestion
    private readonly settings: IPythonSettings;
```

Cuz its initialized in ctor

Reply:
Please change this, we have come across instances where we've used `!` incorrectly and has resulted in bugs (cuz we forget about this during refactoring or in cases its just human error - during runtime it ends up being null and then falls over).